python/triton/code_gen.py

```diff
@@ -518,6 +523,16 @@ def __call__(self, stream, args, grid_0, grid_1=1, grid_2=1):
                                 self.bin.num_warps * 32, 1, 1, 
                                 args, self.bin.shared_mem)
 
+    def get_sass(self, fun=None):
+        if self.sass:
+            return self.sass
+        fd, path = tempfile.mkstemp()
```

Comment:
is it possible to use a contextmanager here? otherwise I feel like this temporary file won't get removed if `cubin.write` or `extract` raises an exception